### PR TITLE
[nav] bio-inspired navigation module

### PR DESCRIPTION
**Pull request by gautierhattenberger**
based on the preliminary work of an internship
integrate a bio-inspired navigation based on the behavior of fish

require the message from https://github.com/paparazzi/pprzlink/pull/130

### CODE REVIEW

File: conf/flight_plans/ENAC/fish_outdoor.xml [hunk 1]
```diff
@@ -0,0 +1,115 @@
+<!DOCTYPE flight_plan SYSTEM "../flight_plan.dtd">
+
+<flight_plan alt="190" ground_alt="185" lat0="43.462976" lon0="1.2733345" max_dist_from_home="50" name="Demo fish nav Muret outdoor" security_height="0.3">
+  <header>
+#include "autopilot.h"
+#include "subsystems/datalink/datalink.h"
```
Comment (noether) [inline conf/flight_plans/ENAC/fish_outdoor.xml:6]: Just curiosity, why is this necessary here? Because of the exceptions in the flightplan?

File: conf/modules/nav_fish.xml [hunk 2]
```diff
@@ -0,0 +1,47 @@
+<!DOCTYPE module SYSTEM "module.dtd">
+
+<module name="nav_fish" dir="nav">
+  <doc>
+    <description>
+      Bio-inspired swarm navigation
```
Comment (noether) [inline conf/modules/nav_fish.xml:6]: If available, a link to HAL or similar to the algorithm description?

File: sw/airborne/modules/nav/nav_fish.c [hunk 3]
```diff
@@ -0,0 +1,505 @@
...
+      &nav_fish.f_ali,
+      &nav_fish.f_att);
+}
+
+/** sign function
+ *
+ * @param number x
+ * @return sign of x
+ */
+static float sign(float x)
```
Comment (noether) [inline sw/airborne/modules/nav/nav_fish.c:232]: I do not know whether is necessary here. 

Sometimes, this sign function implementation produces chattering (jumping between -1 and 1 all the time) if we are close to zero (also I do not know whether you had any issue in your flights). I do not say you must implement it here since I do not know how easy is to implement hysteresis. Probably, it is something worth to implement in paparazzi_math xD.

**Review by noether — APPROVED** [review]
I have some small comments, but the PR looks clear, and it can be taken as a good template for other guys.

File: conf/flight_plans/ENAC/fish_outdoor.xml (hunk 1 above)
Comment (gautierhattenberger) [reply conf/flight_plans/ENAC/fish_outdoor.xml:6]: yes, for `datalink_time` variable

File: sw/airborne/modules/nav/nav_fish.c (hunk 3 above)
Comment (gautierhattenberger) [reply sw/airborne/modules/nav/nav_fish.c:232]: The function using this might change anyway, so I'm not sure if we will keep it in the end. But I agree that this could be integrated someday to the math lib, as well as an hysteresis function.

File: conf/modules/nav_fish.xml (hunk 2 above)
Comment (gautierhattenberger) [reply conf/modules/nav_fish.xml:6]: done
